File: frontend/src/types/transaction.ts
export interface Transaction {
  id: number;
  amount: number;
  merchantName: string;
  tenpistName: string;
  transactionDate: string;
  createdAt: string;
}

export interface TransactionRequest {
  amount: number;
  merchantName: string;
  tenpistName: string;
  transactionDate: string;
}

export interface ApiError {
  timestamp: string;
  status: number;
  error: string;
  message: string;
  path: string;
}

File: backend/src/main/java/com/tenpo/transactions/entity/Transaction.java
package com.tenpo.transactions.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "transactions")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Transaction {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private Integer amount;

    @Column(name = "merchant_name", nullable = false, length = 255)
    private String merchantName;

    @Column(name = "tenpist_name", nullable = false, length = 255)
    private String tenpistName;

    @Column(name = "transaction_date", nullable = false)
    private LocalDateTime transactionDate;

    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        this.createdAt = LocalDateTime.now();
    }
}
